packages/webapp/src/hooks/api/useMutations.js

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+const useMutations = (namedMutations) => {
+  const results = namedMutations.map((mutation) => mutation.hook());
+  const data = results.map((result) => result[1]);
+
+  const isError = data.some((result) => result.isError);
+  const isLoading = data.some((result) => result.isLoading) && !isError;
+  const isSuccess = data.every((result) => result.isSuccess);
```

Comment:
I just realized that when only some of the mutations are actually called, even if all of them are successful, `isSuccess` won't be true.
For instance, we pass removeAnimals, removeBatches, deleteAnimals, and deleteBatches, but we will call only two of them at most (either removeAnimals and removeBatches, or deleteAnimals and deleteBatches) at the same time. I expect `isSuccess` to be true if all the APIs I called were successful, but it's not the case because the other APIs aren't called and `isSuccess` of their results won't be `true`.

Is there a solution?

Reply:
Ah this is a really good point! I think to be able to leverage `isSuccess` and `isError` correctly we'd have to call `useMutations` twice, once for deletion and once for removal, and handle the states separately

Reply:
Thanks for spotting! I added the remove part after un-drafting and did not think that through!

Reply:
I think I will remove those from the useMutations since as you mention the nature of mutations makes the likelihood of simultaneous triggers of all unlikely. Just use it for the naming convenience for now 😂

Reply:
For more info -- I think I would have to call it four times actually in this case haha -- since we conditionally call the animals or batches if there are any selected. So it would probably defeat the purpose of an organizing hook.